value_source.go

```diff
@@ -17,6 +17,11 @@ type ValueSource interface {
 	Lookup() (string, bool)
 }
 
+type EnvValueSource interface {
+	IsFromEnv() bool
```

Comment:
LGTM. Only 1 comment.

I'm wondering about the use cases for `IsFromEnv`. Doesn't it always return true?

Reply:
Yes does in our case as it enables us to iterate over value sources and detect which ones are from env (for help print purposes). It is conceivable for us to have an envSource which masks IsFromEnv so that it doesnt show up in help. Think of boolFlag pattern. 

https://cs.opensource.google/go/go/+/refs/tags/go1.23.3:src/flag/flag.go;l=150